### ISSUE TITLE
Null Pointer Exception on Android 9/10 on Samsung Galaxy Note using SDK version 3.0.13 - Cordova

### ISSUE THREAD

**Issue by woody0** [thegrizzlylabs/geniusscan-sdk-demo#24]
Samsung Galaxy Note10+ (d2q), 11520MB RAM, Android 9

java.lang.RuntimeException: 
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4063)
  at android.app.ActivityThread.handleResumeActivity (ActivityThread.java:4095)
  at android.app.servertransaction.ResumeActivityItem.execute (ResumeActivityItem.java:51)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:145)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:70)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1994)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:216)
  at android.app.ActivityThread.main (ActivityThread.java:7263)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:494)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:975)

Caused by: java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4707)
  at android.app.ActivityThread.performResumeActivity (ActivityThread.java:4052)

Caused by: java.lang.NullPointerException: 
  at com.thegrizzlylabs.geniusscan.sdk.cordova.GeniusScan.onActivityResult (GeniusScan.java:108)
  at org.apache.cordova.CordovaInterfaceImpl.onActivityResult (CordovaInterfaceImpl.java:159)
  at org.apache.cordova.CordovaActivity.onActivityResult (CordovaActivity.java:361)
  at android.app.Activity.dispatchActivityResult (Activity.java:7809)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4700)

**Comment by guillaume-tgl** [comment]
- Could you give us more details on when this issue happens? Every time you use our Cordova plugin? When doing specific steps?
- Can you reproduce it with our Cordova demo app?

**Comment by woody0** [comment]
Good morning Guillaume,

The problem seems to be happening on application load, and before any GS functions are called.

Unfortunately, this seems reproducible only on that specific hardware as we have been unable to reproduce it on anything else. We don’t happen to have one of those test devices, so are relying on google and end user reports.

Last night we deployed a new android version with the .14 version of the SDK, and will report if it continues.

Thanks,
                  Jeff

(561) 472-1401

**Comment by woody0** [comment]
Hi Guillaume,

After updating to SDK version 3.0.14, we contacted 2 of our end users that were experiencing the problem, and it is still continuing to happen.

The app crashes when loading, and before any calls to GS are made, the stack trace posted earlier was reported through the Google Play store.

Thanks,
                  Jeff

(561) 472-1401

From: Jeff Woodard
Sent: Tuesday, April 21, 2020 8:01 AM
To: thegrizzlylabs/geniusscan-sdk-demo <reply@reply.github.com>; thegrizzlylabs/geniusscan-sdk-demo <geniusscan-sdk-demo@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [thegrizzlylabs/geniusscan-sdk-demo] Null Pointer Exception on Android 9/10 on Samsung Galaxy Note using SDK version 3.0.13 - Cordova (#24)

Good morning Guillaume,

The problem seems to be happening on application load, and before any GS functions are called.

Unfortunately, this seems reproducible only on that specific hardware as we have been unable to reproduce it on anything else. We don’t happen to have one of those test devices, so are relying on google and end user reports.

Last night we deployed a new android version with the .14 version of the SDK, and will report if it continues.

Thanks,
                  Jeff

(561) 472-1401

**Comment by woody0** [comment]
Hi,

Just wanted to let you know that we saw version 3.0.15 posted, and tried it in our app.

Unfortunately, we’re now getting an error “Alert PDF generation failed with code FILENOTFOUND” on all of our Android test phones.

Thanks,
                  Jeff

(561) 472-1401

From: Jeff Woodard
Sent: Tuesday, April 21, 2020 12:38 PM
To: 'thegrizzlylabs/geniusscan-sdk-demo' <reply@reply.github.com>; 'thegrizzlylabs/geniusscan-sdk-demo' <geniusscan-sdk-demo@noreply.github.com>
Cc: 'Author' <author@noreply.github.com>; Sergio Escalera <SEscalera@siboneycc.com>; 'sdk@geniusscan.com' <sdk@geniusscan.com>
Subject: RE: [thegrizzlylabs/geniusscan-sdk-demo] Null Pointer Exception on Android 9/10 on Samsung Galaxy Note using SDK version 3.0.13 - Cordova (#24)

Hi Guillaume,

After updating to SDK version 3.0.14, we contacted 2 of our end users that were experiencing the problem, and it is still continuing to happen.

The app crashes when loading, and before any calls to GS are made, the stack trace posted earlier was reported through the Google Play store.

Thanks,
                  Jeff

(561) 472-1401

From: Jeff Woodard
Sent: Tuesday, April 21, 2020 8:01 AM
To: thegrizzlylabs/geniusscan-sdk-demo <reply@reply.github.com<mailto:reply@reply.github.com>>; thegrizzlylabs/geniusscan-sdk-demo <geniusscan-sdk-demo@noreply.github.com<mailto:geniusscan-sdk-demo@noreply.github.com>>
Cc: Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: RE: [thegrizzlylabs/geniusscan-sdk-demo] Null Pointer Exception on Android 9/10 on Samsung Galaxy Note using SDK version 3.0.13 - Cordova (#24)

Good morning Guillaume,

The problem seems to be happening on application load, and before any GS functions are called.

Unfortunately, this seems reproducible only on that specific hardware as we have been unable to reproduce it on anything else. We don’t happen to have one of those test devices, so are relying on google and end user reports.

Last night we deployed a new android version with the .14 version of the SDK, and will report if it continues.

Thanks,
                  Jeff

(561) 472-1401
